### PR TITLE
IBX-154: Allowed to define if the alternative text for image field is required (2.5)

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [IBX-154](https://jira.ez.no/browse/IBX-154)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.5`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

https://github.com/ezsystems/ezplatform-kernel/pull/194 rebased to 2.5

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [X] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php [hunk 1]
```diff
@@ -163,6 +166,30 @@ public function toFieldValue(StorageFieldValue $value, FieldValue $fieldValue)
         $fieldValue->data = $this->parseLegacyXml($value->dataText);
     }
 
+    public function toStorageFieldDefinition(FieldDefinition $fieldDef, StorageFieldDefinition $storageDef): void
+    {
+        $validators = $fieldDef->fieldTypeConstraints->validators;
+
+        $storageDef->dataInt1 = $validators['FileSizeValidator']['maxFileSize'] ?? 0;
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php:173]: don't you need to cast that as well?

**Review by alongosz — APPROVED** [review]

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php (hunk 1 above)
Comment (Steveb-p) [reply eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php:173]: I believe it's a backport, so if there is a need for a type case, it should be added to original source as well.

File: eZ/Publish/Core/Persistence/Tests/FieldValue/Converter/ImageConverterTest.php [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+final class ImageConverterTest extends TestCase
+{
+    /** @var \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\ImageConverter */
+    private $imageConverter;
+
+    /** @var \eZ\Publish\Core\IO\UrlRedecoratorInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $urlRedecorator;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    private $ioService;
```
Comment (Steveb-p) [inline eZ/Publish/Core/Persistence/Tests/FieldValue/Converter/ImageConverterTest.php:28]: I think that `MockObject` can be dropped from those for the purposes of future phpstan introduction - they are not configured anywhere.

**Review by Steveb-p — APPROVED** [review]

**Review by ViniTou — APPROVED** [review]

File: eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php (hunk 1 above)
Comment (adamwojs) [reply eZ/Publish/Core/Persistence/Legacy/Content/FieldValue/Converter/ImageConverter.php:173]: > don't you need to cast that as well? 

According to validation schema `$validators['FileSizeValidator']['maxFileSize']` is an integer value (or null) https://github.com/ezsystems/ezpublish-kernel/blob/15ca50fc5d96743080d9d96764531f8a7d7b5ea5/eZ/Publish/Core/FieldType/Image/Type.php#L25-L38

**Review by pbisaga — APPROVED** [review]
QA approved, checked on eZ Platform Enterprise 2.5 with .diff

**Pull request merged**